### PR TITLE
Limit number of linker jobs on arm to avoid OOM during build

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [x]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit c7180e67bde8fa7924c77b999a0dafdc2b4b283d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66435/c7180e67bde8fa7924c77b999a0dafdc2b4b283d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad21a9307fc1d0715388923d8f1e490a7bd9bebe/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by tavplubix — APPROVED** [review]

File: cmake/limit_jobs.cmake [hunk 1]
```diff
@@ -42,9 +42,14 @@ endif ()
 # But use 2 parallel jobs, since:
 # - this is what llvm does
 # - and I've verfied that lld-11 does not use all available CPU time (in peak) while linking one binary
-if (CMAKE_BUILD_TYPE_UC STREQUAL "RELWITHDEBINFO" AND ENABLE_THINLTO AND PARALLEL_LINK_JOBS GREATER 2)
-    message(STATUS "ThinLTO provides its own parallel linking - limiting parallel link jobs to 2.")
-    set (PARALLEL_LINK_JOBS 2)
+if (CMAKE_BUILD_TYPE_UC STREQUAL "RELWITHDEBINFO" AND ENABLE_THINLTO)
+    if (ARCH_AARCH64)
```
Comment (alexey-milovidov) [inline cmake/limit_jobs.cmake:46]: Missing explanation.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by antonio2368** [comment]
Seems like it didn't help https://github.com/ClickHouse/ClickHouse/actions/runs/9935217428/job/27441008909

**Comment by alexey-milovidov** [comment]
@antonio2368 Let's disable LTO on ARM.

**Comment by nickitat** [comment]
Let's try another fix first

https://github.com/ClickHouse/ClickHouse/pull/66530

**Pull request merged**